bioptim/interfaces/acados_interface.py

```diff
@@ -260,8 +260,13 @@ def __update_solver(self):
                 self.ocp_solver.constraints_set(n, "lbx", self.x_bound_min[:, 1])
                 self.ocp_solver.constraints_set(n, "ubx", self.x_bound_max[:, 1])
 
-        if self.y_ref_end:
-            self.ocp_solver.cost_set(self.acados_ocp.dims.N, "yref", np.concatenate([data for data in self.y_ref_end]))
+        if self.y_ref:
+            if self.y_ref_end:
+                self.ocp_solver.cost_set(self.acados_ocp.dims.N, "yref",
+                                         np.concatenate([data for data in self.y_ref_end]))
+            else:
+                self.ocp_solver.cost_set(self.acados_ocp.dims.N, "yref",
```

Comment:
Is that intended that if y_ref_end exists that y_ref is therefore completely skipped?

Reply:
I think it is not. But currently this is how acados seems to be interfaced. I'm investigating this issue !